app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java

```diff
@@ -1368,10 +1368,23 @@ public boolean requestApkInstallation(final Context context, final File file) {
         }
 
         if (fileUri != null) {
-            final Intent intent = new Intent(Build.VERSION.SDK_INT >= Build.VERSION_CODES.N ? Intent.ACTION_INSTALL_PACKAGE : Intent.ACTION_VIEW)
-                    .setFlags(Build.VERSION.SDK_INT >= Build.VERSION_CODES.N ? Intent.FLAG_GRANT_READ_URI_PERMISSION : Intent.FLAG_ACTIVITY_NEW_TASK)
-                    .setDataAndType(fileUri, "application/vnd.android.package-archive");
-            startActivity(context, intent);
+            final String MIME_TYPE_APK = "application/vnd.android.package-archive";
+
+            boolean hasRequestInstallPackagesPermission = true;
+            try {
```

Comment:
Note: I tried with PackageManager to check permission granted, but it would not work for this case. When the permission is defined or not in the Manifest the check would always result same.

However, canRequestPackageInstalls() always throws a exception when permission not present in manifest. And not throws one if it's present. Thus this used for detecting permission granted.